modules/container_runtime/main.tf

```diff
@@ -65,5 +68,172 @@ resource "aws_ssm_parameter" "ssm_container_runtime_kms_key_id" {
 resource "aws_ssm_parameter" "ecs_cluster_name" {
   name  = "/${var.environment_name}/${var.container_runtime_name}/container_runtime_ecs_cluster_name"
   type  = "String"
-  value = aws_ecs_cluster.fargate_cluster.name
+  value = local.create_ecs_with_fargate == true ? aws_ecs_cluster.fargate_cluster[0].name : aws_ecs_cluster.ec2_cluster[0].name
+}
+
+# ---------------------------------------------------------------------------------------------------------------------
+# ECS with EC2 instances
+# ---------------------------------------------------------------------------------------------------------------------
+# tfsec:ignore:aws-ecs-enable-container-insight
+resource "aws_ecs_cluster" "ec2_cluster" {
+  count = local.create_ecs_with_ec2 ? 1 : 0
+  name  = var.container_runtime_name
+
+  configuration {
+    execute_command_configuration {
+      logging = "NONE"
+    }
+  }
+}
+
+locals {
+  user_data = <<-EOT
+    #!/bin/bash
+    cat <<'EOF' >> /etc/ecs/ecs.config
+    ECS_CLUSTER=${var.container_runtime_name}
+    ECS_LOGLEVEL=debug
+    EOF
+  EOT
+}
+
+resource "aws_ecs_cluster_capacity_providers" "ecs_ec2_cap_provider" {
+  count              = local.create_ecs_with_ec2 ? 1 : 0
+  cluster_name       = aws_ecs_cluster.ec2_cluster[0].name
+  capacity_providers = [aws_ecs_capacity_provider.ecs_ec2_capacity_provider[0].name]
+
+  default_capacity_provider_strategy {
+    base              = 20
+    weight            = 60
+    capacity_provider = aws_ecs_capacity_provider.ecs_ec2_capacity_provider[0].name
+  }
+}
+
+resource "aws_ecs_capacity_provider" "ecs_ec2_capacity_provider" {
+  count = local.create_ecs_with_ec2 ? 1 : 0
+  name  = "ec2_capacity_provider"
```

Comment:
Rename into terra3_ec2_capacity_provider to more easily identify terra3 resources.

Reply:
done.